enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlService.java

```diff
@@ -49,7 +51,7 @@ public WsdlService() {
     }
     
     public List<WsdlPort> getPorts() {
-        List wsdlPorts = new ArrayList();
+        List<WsdlPort> wsdlPorts = new ArrayList<>();
```

Comment:
ho about rewriting this funtion into

public List<WsdlPort> getPorts() {
        
        if (service==null) {
            return emptyList();
        } else {
            return service.getPorts().stream().map(WsdlPort::new).collect(toList());
        }
}